docs/api_objectives.rst

```diff
@@ -0,0 +1,226 @@
+    :toctree: _api/objectives
+    :recursive:
+    :template: class.rst
+
+    desc.objectives.BoundaryError
+    desc.objectives.VacuumBoundaryError
+
+
+Profiles
+--------
+.. autosummary::
+    :toctree: _api/objectives
+    :recursive:
+    :template: class.rst
+
+    desc.objectives.RotationalTransform
+    desc.objectives.Shear
+    desc.objectives.ToroidalCurrent
+    desc.objectives.Pressure
+    desc.objectives.BootstrapRedlConsistency
+
+
+Fixing degrees of freedom
+-------------------------
+.. autosummary::
+    :toctree: _api/objectives
+    :recursive:
+    :template: class.rst
+
+    desc.objectives.FixBoundaryR
+    desc.objectives.FixBoundaryZ
+    desc.objectives.FixAxisR
+    desc.objectives.FixAxisZ
+    desc.objectives.FixPsi
+    desc.objectives.FixPressure
+    desc.objectives.FixIota
+    desc.objectives.FixCurrent
+    desc.objectives.FixAtomicNumber
+    desc.objectives.FixElectronDensity
+    desc.objectives.FixElectronTemperature
+    desc.objectives.FixIonTemperature
+    desc.objectives.FixModeR
+    desc.objectives.FixModeZ
+    desc.objectives.FixSumModesR
+    desc.objectives.FixSumModesZ
+    desc.objectives.FixThetaSFL
+    desc.objectives.FixParameter
+
+
+User defined objectives
+-----------------------
+.. autosummary::
+    :toctree: _api/objectives
+    :recursive:
+    :template: class.rst
+
+
+    desc.objectives.GenericObjective
+    desc.objectives.ObjectiveFromUser
+    desc.objectives.LinearObjectiveFromUser
+
+
+Utilities for getting common groups of constraints
+--------------------------------------------------
+.. autosummary::
+    :toctree: _api/objectives
+    :recursive:
+    :template: class.rst
+
+    desc.objectives.get_fixed_boundary_constraints
+    desc.objectives.get_NAE_constraints
+    desc.objectives.get_fixed_axis_constraints
+    desc.objectives.get_equilibrium_objective
+
+
+Optimization
+************
+``desc.optimize.Optimizer`` is the primary interface, it contains wrappers for a number
+of different methods listed in `Optimizers Supported <https://desc-docs.readthedocs.io/en/latest/optimizers.html>`_.
+
+.. autosummary::
+   :toctree: _api/optimize
+   :recursive:
+   :template: class.rst
+
+   desc.optimize.Optimizer
+
+There are also a number of optimizers written specifically for DESC that we also offer
+with a direct interface similar to ``scipy.optimize.minimize``:
+
+.. autosummary::
+   :toctree: _api/optimize
+   :recursive:
+   :template: class.rst
+
+   desc.optimize.lsqtr
+   desc.optimize.fmintr
+   desc.optimize.fmin_auglag
+   desc.optimize.lsq_auglag
+   desc.optimize.sgd
+
+DESC also allows you to use custom optimizers by creating a wrapper function and
+registering it using ``desc.optimize.register_optimizer``. See `Adding optimizers <https://desc-docs.readthedocs.io/en/stable/adding_optimizers.html>`_
+for details
+
+.. autosummary::
+   :toctree: _api/optimize
+   :recursive:
+   :template: class.rst
+
+   desc.optimize.register_optimizer
+
+
+Grids
```

Comment:
Do the grids belong in the "objectives" file?

Reply:
I figured that's how most users will interact with them - by picking a grid to evaluate an objective on. Not sure where else to put them. I deliberately didn't include Basis/Transform stuff in any of the new ones since those are more internal (though still documented in the alphabetical listing)